### PR TITLE
feat(explore-toolbar): Adding search to groubBy and delete button to rows

### PR DESCRIPTION
**Pull request by Abdkhan14**
Screenshot 2024-09-06 at 3 18 58 PM

https://github.com/user-attachments/assets/fcbc0fb9-22d1-4f61-91db-f3a18af4d5b9

### CODE REVIEW

File: static/app/views/explore/toolbar/styles.tsx [hunk 1]
```diff
@@ -28,3 +31,47 @@ export const ToolbarHeading = styled('h6')<{disabled?: boolean}>`
...
+      const newRow = rows.filter((_, rowIndex) => rowIndex !== i);
+      setRows?.(newRow);
+    },
+    [setRows, rows]
+  );
+
+  return (
+    <ToolbarRowWrapper>
+      {children}
+      {rows.length > 1 ? (
```
Comment (Zylphrex) [inline static/app/views/explore/toolbar/styles.tsx:57]: Instead of hiding, we should disable the button with `disable={rows.length <= 1}`

File: static/app/views/explore/toolbar/toolbarVisualize.tsx [hunk 2]
```diff
@@ -79,7 +79,7 @@ export function ToolbarVisualize({}: ToolbarVisualizeProps) {
         </ToolbarHeaderButton>
       </ToolbarHeader>
       {parsedVisualizes.map((parsedVisualize, index) => (
-        <VisualizeOption key={index}>
+        <ToolbarRow rows={visualizes} setRows={setVisualizes} index={index} key={index}>
```
Comment (Zylphrex) [inline static/app/views/explore/toolbar/toolbarVisualize.tsx:82]: Can you undo the changes to this file here? We're discussing some changes that would be incompatible with this helper component.

Comment (Abdkhan14) [reply static/app/views/explore/toolbar/toolbarVisualize.tsx:82]: Okay so will just keep the delete to groupBy for now

File: static/app/views/explore/toolbar/styles.tsx (hunk 1 above)
Comment (Abdkhan14) [reply static/app/views/explore/toolbar/styles.tsx:57]: Sure yup

**Review by Zylphrex — APPROVED** [review]
